### DISCUSSION TITLE
GAMA integrate with Python ?

### DISCUSSION THREAD

**Discussion by trungthinhnguyen** [gama-platform/gama#375] [Q&A] [answered]
I am currently having a code in python that i use to train robot model moving in basement. I want to use it to visualize in GAMA. I have read about GAMA server but I am not sure it will match with my demand

**Comment by lesquoyb** [comment]
I don't really understand your question, could you be more precise ? What do you want to do exactly and what do you think could be a problem ?

**Reply by trungthinhnguyen** [reply]
I am training a robot that run in python library. I have exported 2 csv files. The first one is grid map (include -1, 0; -1 means obstacles and 0 is free path). I don't know how can i create that grid map in GAMA.

**Reply by trungthinhnguyen** [reply]
[grid_state.csv](https://github.com/user-attachments/files/17986239/grid_state.csv)
you can check the format of csv file here

**Reply by trungthinhnguyen** [reply]
![z6090342632288_f863b2f966f5b5753e5b77a44365baa9](https://github.com/user-attachments/assets/55fd33bf-0bb3-4557-9782-f7b73ec7e47d)
this is the output i have visualized using pygame. I want GAMA to do the same for me

**Reply by lesquoyb** [reply, marked as answer]
Hello, this is a bit tricky to do for a beginner so I provided an example here:
```
/**
* Name: main
* Based on the internal empty template. 
* Author: baptiste
* Tags: 
*/


model main

global{
	
	file my_file <- csv_file('../includes/grid_state.csv', ',');
	
	init {
		matrix<int> data <- matrix<int>(my_file);
		loop x from: 0 to:data.columns-1 {
			loop y from:0 to:data.rows-1 {
				my_grid[x, y].value <- data[x,y];
				switch data[x, y] {
					match 0 {
						my_grid[x, y].color <- #white;				
					}
					match -1 {
						my_grid[x, y].color <- #black;
					}
					match 1 {
						my_grid[x, y].color <- #blue;
					}
				}
			}
		}
	}
}

grid my_grid width:40 height:40{
	
	int value;

}

experiment testing {
	output{
		display main_grid type:2d antialias:false{
			grid my_grid border:#black;
		}
	}
}
```
This is assuming your csv file is stored in the includes folder, is of fixed size 40x40 and doesn't change.
If your file is going to change then you will need to move a bit the code for it to be executed in reflexes instead of in the init blocks.
I advise that you follow the tutorial [here](https://gama-platform.org/wiki/next/PredatorPrey) which is short and will showcase many important gama features step by step and especially grid manipulation.

**Reply by trungthinhnguyen** [reply]
Sorry for late appreciation, i have done it. Thanks you
